### DISCUSSION TITLE
Reload the UI everytime a file is changed

### DISCUSSION THREAD

**Discussion by xpierroz** [zauberzeug/nicegui#996] [Q&A]
### Question

Hey, 
so let's say I have a label, using for text the content of my "helloworld.txt" file
In my helloworld.txt file, I have "Hello world!"

However, I want to create something, that will, whenever the helloworld.txt file is changed, if my NiceGUI app is running, then it has to edit the label to use the new text, how can I create an event like that?

**Comment by rodja** [comment]
You can use [watchfiles](https://pypi.org/project/watchfiles/) to get a callback about file changes. Then simply hook it to a function which reads the content and set the new text on the label.
